cuda_core/cuda/core/experimental/_launcher.py

```diff
@@ -34,7 +35,7 @@ def _lazy_init():
     _inited = True
 
 
-def launch(stream, config, kernel, *kernel_args):
+def launch(stream: Union[Stream, IsStreamT], config: LaunchConfig, kernel: Kernel, *kernel_args):
```

Comment:
It's a bit inconsistent where only this API supports `__cuda_stream__` protocol, but many other APIs only work with explicit `Stream` type objects. We should probably be consistent.

Maybe push this to a follow up PR?